tests/virt_autotest/virtual_network_utils.pm

```diff
@@ -0,0 +1,136 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+package virtual_network_utils;
+# Summary: virtual_network_utils: The initial version of virtualization automation test in openqa.
+#          This file provides fundamental utilities.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base Exporter;
+use Exporter;
+
+use utils;
+use strict;
+use warnings;
+use File::Basename;
+use testapi;
+use Data::Dumper;
+use XML::Writer;
+use IO::File;
+use proxymode;
+use version_utils 'is_sle';
+
+our @EXPORT
+  = qw(prepare_network restore_standalone destroy_standalone restart_libvirtd restore_guests restore_network destroy_vir_network restore_libvirt_default);
+
+sub prepare_network {
+    #Confirm the host bridge configuration file
+    my $virt_host_bridge = shift;
+    my $config_path      = "/etc/sysconfig/network/ifcfg-$virt_host_bridge";
+    if (script_run("[[ -f $config_path ]]") != 0) {
+        assert_script_run("ip link add name $virt_host_bridge type bridge");
+        assert_script_run("ip link set dev $virt_host_bridge up");
+        my $bridge_setup = <<'EOF';
+#!/bin/bash
+. /etc/os-release
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='none'\nSTARTMODE='auto'"
+bridge="BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'"
+> /etc/sysconfig/network/ifcfg-br0.new
+echo -e $bridge >/etc/sysconfig/network/ifcfg-br0
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-br0
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+if [[ $VERSION_ID =~ '11' ]]; then
+    service libvirtd restart;
+else
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_setup, 900);
```

Comment:
Why don't you just run those commands openQA natively?

Reply:
There is the background， there was not created host bridge network interface (br0) as default on worker (OSD), so, before trigger libvirt virtual network test, make sure the host bridge network interface(br0) as prepare_network step under virtual_network_utils. pm.

Reply:
Please remove those two lines:
```
my $bridge_destroy = <<'EOF';
#!/bin/bash
```

Reply:
Had updated code changed